test/testcafe/spec/ChallengeCustomAppPush_spec.js

```diff
@@ -0,0 +1,98 @@
+  .respond(success);
+
+const pushWaitMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(pushPoll)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/poll')
+  .respond(pushPoll);
+
+const pushRejectMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(pushPoll)
+  .onRequestTo('http://localhost:3000/idp/idx/challenge/poll')
+  .respond(pushPoll);
+
+fixture('Challenge Custom App Push');
+
+async function setup(t) {
+  const challengeCustomAppPushPageObject = new ChallengeCustomAppPushPageObject(t);
+  await challengeCustomAppPushPageObject.navigateToPage();
+  return challengeCustomAppPushPageObject;
+}
+
+test
+  .requestHooks(pushSuccessMock)('challenge custom app push screen has right labels', async t => {
+    const challengeCustomAppPushPageObject = await setup(t);
+    await checkConsoleMessages({
+      controller: 'mfa-verify',
+      formName: 'challenge-poll',
+      authenticatorKey: 'custom_app',
+      methodType: 'push',
+    });
+
+    const pageTitle = challengeCustomAppPushPageObject.getPageTitle();
+    const pushBtn = challengeCustomAppPushPageObject.getPushButton();
+    const a11ySpan = challengeCustomAppPushPageObject.getA11ySpan();
+    await t.expect(pushBtn.textContent).contains('Push notification sent');
+    await t.expect(a11ySpan.textContent).contains('Push notification sent');
+    await t.expect(pushBtn.hasClass('link-button-disabled')).ok();
+    await t.expect(pageTitle).contains('Get a push notification');
+
+    // Verify links
+    await t.expect(await challengeCustomAppPushPageObject.switchAuthenticatorLinkExists()).ok();
+    await t.expect(challengeCustomAppPushPageObject.getSwitchAuthenticatorLinkText()).eql('Verify with something else');
+    await t.expect(await challengeCustomAppPushPageObject.signoutLinkExists()).ok();
+    await t.expect(challengeCustomAppPushPageObject.getSignoutLinkText()).eql('Back to sign in');
+  });
+
+test
+  .requestHooks(logger, pushSuccessMock)('challenge Custom App push request', async t => {
+    await setup(t);
+    const successPage = new SuccessPageObject(t);
+    const pageUrl = await successPage.getPageUrl();
+    await t.expect(pageUrl)
+      .eql('http://localhost:3000/app/UserHome?stateToken=mockedStateToken123');
+    await t.expect(logger.count(() => true)).eql(1);
+
+    const { request: {
+      body: answerRequestBodyString,
+      method: answerRequestMethod,
+      url: answerRequestUrl,
+    }
+    } = logger.requests[0];
+    const answerRequestBody = JSON.parse(answerRequestBodyString);
+    await t.expect(answerRequestBody).eql({
+      stateHandle: '02XxswQHZqCz4aWbqQFNXYRC4HHoqd76NBC-FIGbbH'
+    });
+    await t.expect(answerRequestMethod).eql('post');
+    await t.expect(answerRequestUrl).eql('http://localhost:3000/idp/idx/challenge/poll');
+  });
+
+test
+  .requestHooks(pushWaitMock)('Warning callout appears after 30 seconds', async t => {
```

Comment:
add a test case for
- error message
- polling error ?
- good to add a test that checks the polling has stopped after error and success remediation to avoid rate limiting issues

Reply:
if the backend has to send this error (based on your comment on the View, Pradeep), then we need to change this test to mock an extra response with that error. This will avoid waiting 30 secs

Reply:
yes. Unless the switch to remediation based error or warning is complete without any flag checks we need to have this test.